vms/platformvm/txs/mempool/mempool.go

```diff
@@ -292,35 +259,24 @@ func (m *mempool) GetDropReason(txID ids.ID) error {
 	return err
 }
 
-func (m *mempool) register(tx *txs.Tx) {
-	txBytes := tx.Bytes()
-	m.bytesAvailable -= len(txBytes)
-	m.bytesAvailableMetric.Set(float64(m.bytesAvailable))
-}
-
-func (m *mempool) deregister(tx *txs.Tx) {
-	txBytes := tx.Bytes()
-	m.bytesAvailable += len(txBytes)
-	m.bytesAvailableMetric.Set(float64(m.bytesAvailable))
-
-	inputs := tx.Unsigned.InputIDs()
-	m.consumedUTXOs.Difference(inputs)
-}
-
 // Drops all [txs.Staker] transactions whose [StartTime] is before
 // [minStartTime] from [mempool]. The dropped tx ids are returned.
 //
 // TODO: Remove once [StartTime] field is ignored in staker txs
-func DropExpiredStakerTxs(mempool Mempool, minStartTime time.Time) []ids.ID {
+func (m *mempool) DropExpiredStakerTxs(minStartTime time.Time) []ids.ID {
```

Comment:
This is not efficient, we have to iterate through the entire mempool. The fix for this is to move verification of the StartTime to the block builder- this will be done in a follow-up.